### PR TITLE
Issue 2217: Integration tests for batch client.

### LINKED ISSUE #2217: Create integration test for the BatchClient
**Problem description**
Right now the BatchClient interface is not covered by any tests. (The impl is)

**Problem location**
Integration tests

**Suggestions for an improvement**
We should add an integration test to cover this code.

### PR DESCRIPTION
**Pull request by shrids**
Signed-off-by: Sandeep <sandeep.shridhar@emc.com>

**Change log description**
Integration tests for batch client.

**Purpose of the change**
Fixes #2217 

**What the code does**
Creates a integration test to validate batch client apis.

**How to verify it**
Tests should pass.

### CODE REVIEW

File: test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        map.put(0.5, 1.0);
+        result = controller.scaleStream(stream, Arrays.asList(1, 2, 3), map, executor).getFuture().get();
+        assertTrue("scale down operation result", result);
+        writeEvents(writer);
+
+        BatchClient batchClient = clientFactory.createBatchClient();
+
+        //List out all the segments in the stream.
+        ArrayList<SegmentInfo> segments = Lists.newArrayList(batchClient.listSegments(stream));
+        assertEquals("Excepted number of segments", 6, segments.size());
```
Comment (fpj) [inline test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java]: `Excepted` -> `Expected`

File: test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        Segment seg0 = new Segment(SCOPE, STREAM, 0);
+        @Cleanup
+        SegmentIterator<String> seg0Iterator = batchClient.readSegment(seg0, serializer, 60);
+        ArrayList<String> dataAtOffset = Lists.newArrayList(seg0Iterator);
+        assertEquals(1, dataAtOffset.size());
+        assertEquals(DATA_OF_SIZE_30, dataAtOffset.get(0));
+    }
+
+    private void writeEvents(EventStreamWriter<String> writer) throws InterruptedException, java.util.concurrent.ExecutionException {
+        writer.writeEvent("0", DATA_OF_SIZE_30).get();
```
Comment (fpj) [inline test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java]: Maybe randomize the keys?

File: test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        serviceBuilder.close();
+        zkTestServer.close();
+    }
+
+    @Test(timeout = 50000)
+    public void testBatchClient() throws Exception {
+        StreamConfiguration config = StreamConfiguration.builder()
+                                                        .scope(SCOPE)
+                                                        .streamName(STREAM)
+                                                        .scalingPolicy(ScalingPolicy.byEventRate(10, 2, 1))
```
Comment (fpj) [inline test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java]: Setting a scaling policy is unnecessary given that you are scaling the stream manually in this test, yes?

File: test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java (hunk 2 above)
Comment (shrids) [reply test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java]: done.

File: test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java (hunk 1 above)
Comment (shrids) [reply test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java]: corrected it.

File: test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java (hunk 3 above)
Comment (shrids) [reply test/integration/src/test/java/io/pravega/test/integration/BatchClientTest.java]: updated it to fixed policy.

**Pull request merged**